noise/README.md

```diff
@@ -210,27 +210,32 @@ The Noise Protocol Framework caters for sending early data alongside handshake
 messages. We leverage this construct to transmit:
 
 1. the libp2p identity key along with a signature, to authenticate each party to
-   the other.
-2. arbitrary data private to the libp2p stack. This facility is not exposed to
-   userland. Examples of usage include streamlining muxer selection.
-
-These payloads MUST be inserted into the first message of the handshake pattern
-**that guarantees secrecy**. In practice, this means that the initiator must not
-send a payload in their first message. Instead, the initiator will send its
-payload in message 3 (closing message), whereas the responder will send theirs
-in message 2 (their only message). It should be stressed, that the second
-message of the handshake pattern has forward secrecy, however the sender has not
-authenticated the responder, so this payload might be sent to any party,
-including an active attacker.
+   the other, and
+2. extensions to the Noise handshake
+
+Note that when sending extensions with the first message of the handshake
+pattern, the data is transmitted unencrypted. When sending the payload in
+message 3 (closing message), for the initiator, and in message 2, for the
+responder, the data will be send encrypted with forward secrecy.
+It should be stressed, that while the second message of the handshake pattern
+has forward secrecy, the sender has not yet authenticated the responder,
+so this payload might be sent to any party, including an active attacker.
 
 When decrypted, the payload contains a serialized [protobuf][protobuf]
 `NoiseHandshakePayload` message with the following schema:
 
 ``` protobuf
+message NoiseExtensions {
+    repeated bytes webtransport_certhashes = 1;
+    optional bytes webrtc_fingerprint = 2;
+    repeated string stream_muxers = 3; 
+}
+
+
 message NoiseHandshakePayload {
-  bytes identity_key = 1;
-  bytes identity_sig = 2;
-  bytes data         = 3;
+  bytes identity_key         = 1;
+  bytes identity_sig         = 2;
+  NoiseExtensions extensions = 4;
```

Comment:
Note that https://github.com/libp2p/specs/pull/412 would make use of the Noise [prologue](https://noiseprotocol.org/noise.html#prologue) feature. This is orthogonal to the early data mechanism. 

Thus far we were not planning to wrap the prologue data in a protobuf. That said, I am not opposed to it.

If I understand the Protobuf schema above correctly, this would require us to have a `identity_key` and `identity_sig` in the prologue encoding. Would we be setting that to an empty byte string? I have to give this more thought but as a first thought, this seems not very clean.

Reply:
I don’t think we’d need this Protobuf at all when using Prologue, since prologues are not sent in the wire, they’re just hashed into the handshake transcript. 
I’ll remote the WebRTC extension entry here.

Orthogonal to this PR: Protobufs are great when sent over the wire. They’re not that great if you’re looking for a unique encoding or your data, which seems to be what we need for prologues. Just using a byte array / string seems to be the better solution for that use case.

Reply:
I don't think we should adopt this for the our prologue use. The reason is that the prologue is very specific to the protocol that's using it. Both sides must know the protocol and agree on what the prologue is before the handshake starts. Therefore it makes more sense to have each protocol (like webrtc) define how the prologue is set since it will only ever be set in the context of the protocol.

Reply:
I removed WebRTC and stream muxer selection from the protobuf. WebRTC won't use early data, but a Noise prologue. For muxer selection, the respective PR (#446) will add that field.